### DISCUSSION TITLE
Faker not configured with desired locale in Database Factories

### DISCUSSION THREAD

**Discussion by JamShady** [laravel/framework#39814] [Q&A] [answered]
The docs state that the Faker locale can be set by adding the `faker_locale` property to `config/app.php`

![image](https://user-images.githubusercontent.com/167395/143886764-ac0472e1-74b7-47b8-94ad-aebf9bfceba2.png)

However, this does not work for references to `$this->faker` because the `Factory::withFaker()` method never actually refers to configured locale.

https://github.com/laravel/framework/blob/606ac0a386c1215b8c472520364b7d86f94afb09/src/Illuminate/Database/Eloquent/Factories/Factory.php#L743

`$this->faker` always sets up faker with a US locale, regardless of what's been configured.

**Comment by JamShady** [comment]
Hi, I'm confused why this bug was converted to a discussion?  Can someone please help me understand? Surely when the documentation and code are doing two different things, and there's (what appears to be) a clear absence of implementation in the code, this is a bug?

In my case, literally injecting the desired locale, ala `return Container::getInstance()->make(Generator::class, ['locale' => 'en_GB']);`, made it behave as desired. I'm not sure how to get the config value from this context so didn't make a PR, but I'm sure it's something trivial for the regular maintainers.

**Reply by JamShady** [reply]
@driesvints sorry to tag you but I wasn't sure if you'd see this otherwise - can you help me understand why this isn't a bug please? Are you sure that maybe this wasn't miscategorised? Cheers

**Reply by driesvints** [reply]
First of all, no issue template was filled in. Other, at first glance this seems like a support request to me. Please try some support channels first (Discussions, Discord, forums, ...) before attempting again. Thanks

**Reply by JamShady** [reply]
No issue template filled in - OK, no problem. I just went to 'create issue from line' and didn't realise, but I can deal with that.

What I don't understand is, you want me to try a support channel to make - literally - non-existent code work, before submitting a (correctly formatted) bug request? I'm not trying to teach my granny to suck eggs, but the issue here seems pretty clear and black & white to me?

**Comment by JamShady** [comment, marked as answer]
Apologies all... this is actually working (there's a lot of magic behind the scenes). The problem was I had 'en_UK' as the locale instead of 'en_GB'
